### ISSUE TITLE
List of tested devices

### ISSUE THREAD

**Issue by tukusejssirs** [IzzySoft/Adebar#7]
As the title suggests, it would be good to know on which devices `adebar` has been tested, esp. for those who just heard about `adebar`.

**Comment by IzzySoft** [comment]
If I'd get any feedback, I could do that – but until now I did not. Leaving this issue open for exaclty this purpose: Everybody can enter his/her own "tested devices" this way, so we hopefully end up with an appropriate list :)

Here are the devices I'm using _Adebar_ with currently:

Phones:
- LG Optimus 4X HD (P880) – Android 4.0.3 & 4.1.2 (stock)
- Motorola Milestone 2 (aka Droid 2) – Android 4.2 & 4.3 (CyanogenMod)
- Samsung Galaxy Ace 2 – Android 4.1.2, stock

Tablets:
- Cat Stargate 2 – Android 4.1.1 (stock)
- Cat PC4 (aka Phoenix) – Android 4.1.1 & 4.2 (stock)

Your turn: next post list at least one MTK device, please :)

**Comment by tukusejssirs** [comment]
So here’s my list (very shooooort one xD):

Phone:
- Lenovo A536 – Android 4.4.2 (stock) rooted
  - [x] deviceInfo.md (for both sim cards), sysApps.md, userApps.md
  - [x] configuration files
    - [x] build.prop, hosts, packages.xml, softap.conf, usage-history.xml, wpa_supplicant.conf<br>
    - [ ] :exclamation: no gps.conf (probably because I never turned it on)
  - [x] TiBu backup
  - [ ] sysbackup
  - [ ] appbackup
  - [ ] TiBu restore
  - [ ] sysrestore
  - [ ] apprestore

Later on I’ll update this post. :)

**Comment by IzzySoft** [comment]
Update: [see here](https://github.com/IzzySoft/Adebar/wiki/Tested-Devices). Linked the page back here a) for details and b) suggesting to add more devices this way. Would be nice to have a little feedback where my little tool is running :innocent:

**Comment by inktrap** [comment]
- Manufacturer: OnePlus
- Model: A0001 (A0001)
- cm-12.1-bacon (YOG4PAO33I)
- tested user/system app backup via adb

**Comment by IzzySoft** [comment]
@inktrap Thanks for the details – but I'd prefered to leave them here, and the wiki-edit to me :) Now I've got to clear a "collusion" as I've made some editing as well: as the wiki has no "branches", I usually push its updates whenever a new version of the code is released; otherwise "new features" already listed in the wiki would cause confusion to people who stick to the releases (well, for the moment we now have to live with that – until the next release is done).

My fault, I should have "locked" the wiki (done now). So some details from your contribution "got lost" – which is why I reverted your edit here. Hope you understand this ;)

**Comment by inktrap** [comment]
No problem! btw, I am running Android 5.1.1, that detail got "lost".

**Comment by inktrap** [comment]
btw, I am running Android 5.1.1, that detail got "lost".

**Comment by IzzySoft** [comment]
> btw, I am running Android 5.1.1, that detail got "lost".

That detail never was part of your post here "in the issue". But it's not lost anyway, as it's part of the wiki page – but feel free to re-edit your above post (click the "pen icon" on its header) to include it there as well :)

**Comment by inktrap** [comment]
:ok_hand:

**Comment by Irfy** [comment]
- Samsung Galaxy Note II (GT-N7100) running KitKat 4.4.2 (stock)
- Huawei Nexus 6P (angler) running Marshmallow 6.0 (rooted)
- Windows/Cygwin
- Tested `{user,sys}{backup,restore}` note2 => nexus6p

Had several issues:
- adb version issues with older Android versions as discussed [here](http://android.stackexchange.com/questions/83080/adb-backup-creates-0-byte-file-prompts-for-current-backup-password-even-though)
  - adb 1.0.32 creates 41-byte backups for _all_ apks
  - solved by downloading adb 1.0.31 (with a couple of dlls) and switcing the binaries for using adebar with note2
  - just quoting arguments to `adb backup` didn't work (see link above)
- `adb` wouldn't understand subdirectory arguments with `-f`
  - solved by creating the backup in CWD and then moving to desired dir.

I don't know if you intend to try to work around the adb issue, but it should be documented, so that the user knows what to try if they get 0-byte, 41-byte or otherwise fixed-size small backup files or all apks on older androids.

**Comment by IzzySoft** [comment]
Thanks for the detailed report, @Irfy ! I've updated the documentation accordingly ([Tested Devices](https://github.com/IzzySoft/Adebar/wiki/Tested-Devices) and [Known Issues](https://github.com/IzzySoft/Adebar/wiki/Known-Issues)). See the latter place for a partial work-around on the ADB issue: you can add `alias adb=/path/to/adb` to your device-specific config files, pointing to the resp. `adb` binary to use.

I'm not sure if I should implement the "argument quoting" (escape_arg) in _Adebar_, e.g. adding a new config var so the user can easily switch it on. Seems you're not the only one hit by this issue (in the SE post, I count at least 3 people with it). Shouldn't be a big deal – feel free opening a separate issue for that.

**Comment by chaotix-** [comment]
- Nexus 7 (2013) LTE (razorg) running Marshmallow 6.0.1 (stock)
- Sony Xperia Z Ultra (togari) running Cyanogenmod 12.1 (20151204) aka Android 5.1.1

(Had an issue with ADB. Doesn't sound like the one described in [Known Issues](https://github.com/IzzySoft/Adebar/wiki/Known-Issues). Bit I fixed it with the patch in PR #15.)

**Comment by IzzySoft** [comment]
Thanks, @chaotix- – just added to the wiki. Will take a look at the other issue ASAP.

**Comment by IzzySoft** [comment]
Also successfully using it now with the _Wileyfox Swift_ running CyanogenOS 12.1 / Android 5.1.1. Device is rooted, no issues found so far.

Thanks to this device I was now able to improve the "DualSIM related details" which I wasn't able to test before, and to add a bunch of new details to the device documentation page :)

**Comment by jreast** [comment]
@IzzySoft I managed to use your script to transfer most of my Apps & data (63 of 82):
Backup from Moto G 2nd Gen [XT1039] - Android 5.1 (Stock not rooted)
Restore to Moto G4 [XT1622] - Android 6.0.1 (Stock not rooted)
I was using cygwin on Windows 7, with the Android SDK (Windows version).
The main thing to watch out for was the git clone converted the scripts to CrLf which breaks them, so be sure to set autocrlf = false before cloning.

Thank you, it really saved me a lot of hassle.

**Comment by IzzySoft** [comment]
Thanks @jreast – glad to read! Just added your device to the Wiki ([Tested Devices](https://github.com/IzzySoft/Adebar/wiki/Tested-Devices)).

As for the CRLF: I assume you've cloned on Windows, and not in Cygwin itself? Thanks for the hint, which I just integrated with the Wiki as well ([Installation](https://github.com/IzzySoft/Adebar/wiki/Installation)).

**Comment by JimboJoe** [comment]
Hello,
This was tested with success under Windows with MobaXterm with:

- Xiaomi Mi4C under OmniROM 6.0.1
- Lenovo/Motorola Moto G4 under stock ROM 6.0.1

Thanks a lot!

**Comment by IzzySoft** [comment]
Thanks @JimboJoe – Wiki updated again :+1:

**Comment by Jakuje** [comment]
Just tested with Coolpad Porto S (E570), Android 5.1 and I am getting  zero-length applications backup files. I tried with ADB 1.0.31, 1.0.32 (the same as on the phone) and 1.0.36 (latest), but without any change. Trying directly `bu 1 backup -apk app.package.name > /sdcard/backup.ab` as advised in [android.SE answer](https://android.stackexchange.com/a/143481/121519) results again in empty backup file, but no errors though. Do you think there is a way how to get around this problem or it is a lost case?

**Comment by IzzySoft** [comment]
@Jakuje that's nothing Adebar can change. Adebar just creates the script for the backup, which then is performed by ADB. You could ask about that issue at Android.SE, hopefully some ideas pop up (please include whether this happens for all apps, or just for specific ones – in the latter case it could simply be they've "opted out", having set `ALLOW_BACKUP:FALSE`). But please, no "thread hi-jacking": This issue here is to report devices successfully running with Adebar, to have them listed as "definitely working with it". In this context: Adebar is working fine with your Coolpad – apart from ADB not being able to produce backups?

**Comment by walkjivefly** [comment]
adebar-cli and userbackup runs on Xiaomi Redmi3 Pro running CM12.1 although lots of 0/41 byte files produced. AUTO_CONFIRM=1 doesn't work. Haven't tried sysbackup or any restores yet.

**Comment by IzzySoft** [comment]
Thanks @walkjivefly – wiki updated! AUTO_CONFIRM indeed seems to have trouble with some devices (I remember one where I wasn't able to get it to work either). As for the 0/41 byte files: not much we can do about that if the app is refusing being backed up. Xposed with BackupAllApps is the only work-around known to me.

**Comment by jreast** [comment]
@IzzySoft Success on Amazon Kindle Fire HD 8 (7th Gen) backup and restore. Thanks again!

**Comment by IzzySoft** [comment]
Thanks, @jreast – what Android version is that? And "stock ROM" I guess?

**Comment by jreast** [comment]
@IzzySoft, yes stock ROM, not rooted, Fire OS 5.6.2.0

**Comment by IzzySoft** [comment]
Thanks again, @jreast – added to [the wiki](https://github.com/IzzySoft/Adebar/wiki/Tested-Devices#tablets) :+1:

**Comment by nerd190** [comment]
Confirmed devices:
- **Samsung Galaxy S5** Android 7.1.2 (AOSP/CAF rom) Rooted.
- **OnePlus 3** stock: Android 8.0 (OxygenOS 5.0.8) Rooted **and...**
- Stock: Android 9.0 (OxygenOS 9.0.6) Rooted
- **Amlogic S912** (Android TV box) stock Android 7.1.2, Rooted.

Hi izzy, thanks for this tool and your Fdroid repo!
I found this tool whilst making my own backup script, great job!

**Comment by IzzySoft** [comment]
Thanks @5c0tt-b0t – and wow, the first TV box reported, great! Wiki page updated. Glad you like it – enjoy Adebar (and my repo :smiley:)!

**Comment by nerd190** [comment]
Two more devices!:
- **OnePlus 5T** stock OxygenOS 9.0.11 (Android 9  - Pie) Rooted.
- **Rockchip RK3328** stock Android 9 Pie, Rooted.

(Another TV box, different CPU, safe to say it works on the two most popular TV box brands!)
I do not own an AllWinner device unfortunately (the third most popular brand) but Amlogic/Rockchip make up around 80% of all TV boxes out there anyway! :D

**Comment by IzzySoft** [comment]
Thanks again! Added to the wiki. Great to know that most Android TV boxes should be covered as well! Though I'm not sure whether that can be said based on the chip alone. Also depends on their ROM…

**Comment by nerd190** [comment]
I work with Android TV box firmware  (just for fun, to improve the *usually* bad chinese firmware found in *most* of them) unlike Phones they are **very** particular with paths (directories) so do not tend to differ with ROM as the dev **has** to use the standard paths found in the stock ROM, luckily they are very AOSP in that sense, they only differ slightly on /root paths.
I *assume* AllWinner will work too, only one I would *suspect* a problem *possibly* are MediaTek devices that usually have different data paths, these probably account for **not even** 10% of the TV box market, fairly rare.
Thanks again! I am still *slowly* working towards my own solution (just to get more fluent in bash), when features are complete I would gladly submit a PR here also, **IF** of course, the features are even wanted! (Its pure bash also), I plan to have rooted phones backed up & restored without PC access, with the ability to backup without using external storage by using hard-links! (useful for phones like OnePlus, that do not have SD card slots, and owner *may* not own a type-C to USB OTG adapter, or a computer), also the ability to utilise Termux to add things like: rsync, rclone etc for incremental backups, you could even use python scripts, or utilise proot to backup using a full linux distro on your phone from a terminal (Arch Linux has **the most** programmes with arm64 binaries, they work perfectly!)
Basically all I wanna achieve personally, is the option to not need a PC at all, of course, root would be required.

**Comment by IzzySoft** [comment]
Good to know about the Android TV backgrounds – thanks for sharing! Btw, my Lenovo tablet is a MediaTek device, no issues with that.

And of course, PRs are welcome! I cannot think of new features all for myself and usually only add one when I miss it. Having so many project to work on, my work on Adebar is not that continuous either :laughing:

Your backup idea sounds interesting as well (reminds me of rsyncbackup, which I use on my PC). Curious about your restore approach then. But that's a different issue (this one already is quite long; Github already has problems loading all its JS stuff along in my browser, so preview, auto-complete etc. are already not working; need to split this)…

**Comment by nerd190** [comment]
It will be automatically restored, the script can push the restore script to:

**For most devices**
- /system/etc/init.d

**For Magisk users**
- /data/adb/service.d **or**
- /data/adb/post-fs-data.d

The contents of these directories are launched on boot so the backup will be initiated automatically, no PC, magic happens in the background, could also let the user know when its completely done by outputting a file in internal storage called "RestoreComplete" or "RestoreFailed" (the latter containing any errors).

**P.S** great news on Mediatek! I see no problems arrising with AllWinner.
And **yes!** rsyncbackup without needing to turn the PC on! 👍 great performance for incremental backups as you already know rsync only backs up changed files, not all files.
rsync works great in Termux, also works without it too.
rclone can push backups directly to most cloud services (Mega, DropBox, Google Drive etc or even a FTP/SFTP server!) works great, both with or without Termux.
Bonus... rooted android devices can easily install adb & fastboot (something Google removed in Android 5 😠 ) so you could remotely backup to another Android device, rather than a PC.
